src/System.IO.Ports/src/System/IO/Ports/SerialPort.Linux.cs

```diff
@@ -24,9 +25,9 @@ public static string[] GetPortNames()
                 var entries = di.EnumerateFileSystemInfos(@"*", SearchOption.TopDirectoryOnly);
                 foreach (var entry in entries)
                 {
-                    if (Directory.Exists(sysUsbDir + entry.Name) || File.Exists(entry.FullName + "/device/id"))
+                    if (Directory.Exists(entry.FullName + "/device/tty") || Directory.Exists(sysUsbDir + entry.Name))
```

Comment:
I think this will show too much. For example my ttyS29 would pass the test and you would get list for MANY non-existent devices. That will eventually break test runs. (even if that could be fixed)   
I'm yet to see ttyS device without /device/id. If anything I would relax checks for the embedded serial variants. 
I think there the sysfs/dev entires are more accurate.

Reply:
still testing this

Reply:
seems like hummingboar's ttymxc* are not being detected - will take a look